src/native/managed/cdacreader/src/Legacy/SOSDacImpl.cs

```diff
@@ -256,11 +256,100 @@ int ISOSDacInterface.GetMethodDescData(ulong methodDesc, ulong ip, DacpMethodDes
+                {
+                    CopyNativeCodeVersionToReJitData(
+                        requestedNativeCodeVersion,
+                        activeNativeCodeVersion.Value,
+                        &data->rejitDataRequested);
+                }
 
-                throw new NotImplementedException(); // TODO[cdac]: rejit stuff
+                // Total number of jitted rejit versions
+                int cJittedRejitVersions = rejitContract.GetRejitIds(_target, methodDescHandle.Address).Count();
+                data->cJittedRejitVersions = (uint)cJittedRejitVersions;
+
+                // Reverted ReJitInfos
+                if (rgRevertedRejitData == null)
+                {
+                    // No reverted rejit versions will be returned, but maybe caller wants a
+                    // count of all versions
+                    if (pcNeededRevertedRejitData != null)
+                    {
+                        *pcNeededRevertedRejitData = data->cJittedRejitVersions;
+                    }
+                }
+                else
+                {
+                    // Caller wants some reverted rejit versions.  Gather reverted rejit version data to return
+
+                    // Prepare array to populate with rejitids.  "+ 1" because GetReJITIDs
+                    // returns all available rejitids, including the rejitid for the one non-reverted
+                    // current version.
+                    List<TargetNUInt> reJitIds = rejitContract.GetRejitIds(_target, methodDescHandle.Address).ToList();
+
+                    // Go through rejitids.  For each reverted one, populate a entry in rgRevertedRejitData
+                    uint iRejitDataReverted = 0;
+                    ILCodeVersionHandle activeVersion = nativeCodeContract.GetActiveILCodeVersion(methodDesc);
+                    TargetNUInt activeVersionId = rejitContract.GetRejitId(activeVersion);
+                    for (uint i = 0; (i < reJitIds.Count) && (iRejitDataReverted < cRevertedRejitVersions); i++)
+                    {
+                        ILCodeVersionHandle ilCodeVersion = nativeCodeContract.GetILCodeVersions(methodDesc)
+                            .FirstOrDefault(ilcode => rejitContract.GetRejitId(ilcode) == reJitIds[(int)i],
+                                ILCodeVersionHandle.Invalid);
+
+
+                        if (!ilCodeVersion.IsValid || rejitContract.GetRejitId(ilCodeVersion) == activeVersionId)
+                        {
+                            continue;
+                        }
+
+                        NativeCodeVersionHandle activeRejitChild = nativeCodeContract.GetActiveNativeCodeVersionForILCodeVersion(methodDesc, ilCodeVersion);
+                        CopyNativeCodeVersionToReJitData(
+                            activeRejitChild,
+                            activeNativeCodeVersion.Value,
+                            &rgRevertedRejitData[iRejitDataReverted]);
+
+                        iRejitDataReverted++;
```

Comment:
I don't think this is simple to merge. `i` is the current index in rejitData we are examining. `iRejitDataReverted` is the number of rejitDatas we have added to the output. We need to keep track of this because the API is capped at a value and it is returned as a pointer.